src/main/java/de/bonndan/nivio/output/map/hex/GroupAreaFactory.java

```diff
@@ -26,95 +28,112 @@
      * <p>
      * There is clearly much room for improvement here. It's only that I haven't found a better approach so far.
      *
-     * @param occupied       tiles occupied by items
-     * @param group          the group
-     * @param allVertexHexes a mapping from item to its hex (all, unfiltered)
+     * @param hexesToItems tiles occupied by items
+     * @param group        the group
      * @return all hexes the group consists of (an area)
      */
-    public static Set<Hex> getGroup(Set<Hex> occupied, Group group, Map<Item, Hex> allVertexHexes) {
+    public static Set<Hex> getGroup(BidiMap<Hex, Object> hexesToItems, Group group) {
 
         Set<Item> items = group.getItems();
         Set<Hex> inArea = new HashSet<>();
-        List<Item> connected = new ArrayList<>();
 
         if (!items.iterator().hasNext()) {
             LOGGER.warn("Could not determine group area for group {}", group);
             return inArea;
         }
-        //surround each item
+
+        //build the area by adding paths
+        addPathsBetweenClosestItems(hexesToItems, items, inArea);
```

Comment:
Great! Extracting this into an own method makes everything better readable and understandable 😄